kasa/smart/smartdevice.py

```diff
@@ -476,8 +471,8 @@ def update_from_discover_info(self, info):
 
     async def get_emeter_realtime(self) -> EmeterStatus:
         """Retrieve current energy readings."""
-        _LOGGER.warning("Deprecated, use `emeter_realtime`.")
-        if not self.has_emeter:
+        warn("Deprecated, use `emeter_realtime`.", DeprecationWarning, stacklevel=1)
+        if Module.Energy not in self.modules:
             raise KasaException("Device has no emeter")
         return self.emeter_realtime
 
```

Comment:
Just a side note, but we should probably start dropping previously deprecated APIs?